### ISSUE TITLE
No checks for validity of an application name

### ISSUE THREAD

**Issue by bg6954** [cloudfoundry-attic/cf#54]
There seems to be no checks in place for application name, invalid application name is initially accepted with a successful push and start but no way to access the URL.

$ cf push --command "node nodejsHelloworld.js"
Name> nodejsHelloworld0.0.1

Save configuration?> n

Uploading nodejsHelloworld0.0.1... OK

nodejsHelloworld0.0.1 is currently stopped, start it with 'cf start'

$ cf start nodejsHelloworld0.0.1
Preparing to start nodejsHelloworld0.0.1... OK
Checking status of app 'nodejsHelloworld0.0.1'...
  5 of 5 instances running (5 running)
Push successful! App 'nodejsHelloworld0.0.1' available at http://.

**Comment by xoebus** [comment]
Agreed, app names/subdomains should be validated in a way that gives better info back to the user rather than just not starting the app.
